diff --git a/Makefile.am b/Makefile.am
index 29b8a69..8909287 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,10 +4,9 @@ ACLOCAL_AMFLAGS = -I build-aux/m4
 # which does not have an explicit foo_CFLAGS variable set.
 AM_CFLAGS = $(SECP_CFLAGS)
 
-lib_LTLIBRARIES = libsecp256k1.la
-include_HEADERS = include/secp256k1.h
-include_HEADERS += include/secp256k1_preallocated.h
-noinst_HEADERS =
+noinst_LTLIBRARIES = libsecp256k1.la
+noinst_HEADERS = include/secp256k1.h
+noinst_HEADERS += include/secp256k1_preallocated.h
 noinst_HEADERS += src/scalar.h
 noinst_HEADERS += src/scalar_4x64.h
 noinst_HEADERS += src/scalar_8x32.h
@@ -71,7 +70,7 @@ noinst_HEADERS += contrib/lax_der_privatekey_parsing.c
 noinst_HEADERS += examples/examples_util.h
 
 PRECOMPUTED_LIB = libsecp256k1_precomputed.la
-noinst_LTLIBRARIES = $(PRECOMPUTED_LIB)
+noinst_LTLIBRARIES += $(PRECOMPUTED_LIB)
 libsecp256k1_precomputed_la_SOURCES =  src/precomputed_ecmult.c src/precomputed_ecmult_gen.c
 # We need `-I$(top_srcdir)/src` in VPATH builds if libsecp256k1_precomputed_la_SOURCES have been recreated in the build tree.
 # This helps users and packagers who insist on recreating the precomputed files (e.g., Gentoo).
@@ -85,7 +84,7 @@ endif
 noinst_LTLIBRARIES += $(COMMON_LIB)
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = libsecp256k1.pc
+noinst_DATA = libsecp256k1.pc
 
 if USE_EXTERNAL_ASM
 if USE_ASM_ARM
diff --git a/src/modules/ecdh/Makefile.am.include b/src/modules/ecdh/Makefile.am.include
index 7f7f95f..61fc1b0 100644
--- a/src/modules/ecdh/Makefile.am.include
+++ b/src/modules/ecdh/Makefile.am.include
@@ -1,4 +1,4 @@
-include_HEADERS += include/secp256k1_ecdh.h
+noinst_HEADERS += include/secp256k1_ecdh.h
 noinst_HEADERS += src/modules/ecdh/main_impl.h
 noinst_HEADERS += src/modules/ecdh/tests_impl.h
 noinst_HEADERS += src/modules/ecdh/bench_impl.h
diff --git a/src/modules/extrakeys/Makefile.am.include b/src/modules/extrakeys/Makefile.am.include
index 0d901ec..017d6b4 100644
--- a/src/modules/extrakeys/Makefile.am.include
+++ b/src/modules/extrakeys/Makefile.am.include
@@ -1,4 +1,4 @@
-include_HEADERS += include/secp256k1_extrakeys.h
+noinst_HEADERS += include/secp256k1_extrakeys.h
 noinst_HEADERS += src/modules/extrakeys/tests_impl.h
 noinst_HEADERS += src/modules/extrakeys/tests_exhaustive_impl.h
 noinst_HEADERS += src/modules/extrakeys/main_impl.h
diff --git a/src/modules/recovery/Makefile.am.include b/src/modules/recovery/Makefile.am.include
index 156ea69..c8a275b 100644
--- a/src/modules/recovery/Makefile.am.include
+++ b/src/modules/recovery/Makefile.am.include
@@ -1,4 +1,4 @@
-include_HEADERS += include/secp256k1_recovery.h
+noinst_HEADERS += include/secp256k1_recovery.h
 noinst_HEADERS += src/modules/recovery/main_impl.h
 noinst_HEADERS += src/modules/recovery/tests_impl.h
 noinst_HEADERS += src/modules/recovery/tests_exhaustive_impl.h
diff --git a/src/modules/schnorrsig/Makefile.am.include b/src/modules/schnorrsig/Makefile.am.include
index 654fa2e..2fc623c 100644
--- a/src/modules/schnorrsig/Makefile.am.include
+++ b/src/modules/schnorrsig/Makefile.am.include
@@ -1,4 +1,4 @@
-include_HEADERS += include/secp256k1_schnorrsig.h
+noinst_HEADERS += include/secp256k1_schnorrsig.h
 noinst_HEADERS += src/modules/schnorrsig/main_impl.h
 noinst_HEADERS += src/modules/schnorrsig/tests_impl.h
 noinst_HEADERS += src/modules/schnorrsig/tests_exhaustive_impl.h
diff --git a/src/modules/ellswift/Makefile.am.include b/src/modules/ellswift/Makefile.am.include
index 8251231..133fc12 100644
--- a/src/modules/ellswift/Makefile.am.include
+++ b/src/modules/ellswift/Makefile.am.include
@@ -1,4 +1,4 @@
-include_HEADERS += include/secp256k1_ellswift.h
+noinst_HEADERS += include/secp256k1_ellswift.h
 noinst_HEADERS += src/modules/ellswift/bench_impl.h
 noinst_HEADERS += src/modules/ellswift/main_impl.h
 noinst_HEADERS += src/modules/ellswift/tests_impl.h
diff --git a/src/modules/musig/Makefile.am.include b/src/modules/musig/Makefile.am.include
index 796443c..4e1136c 100644
--- a/src/modules/musig/Makefile.am.include
+++ b/src/modules/musig/Makefile.am.include
@@ -1,4 +1,4 @@
-include_HEADERS += include/secp256k1_musig.h
+noinst_HEADERS += include/secp256k1_musig.h
 noinst_HEADERS += src/modules/musig/main_impl.h
 noinst_HEADERS += src/modules/musig/keyagg.h
 noinst_HEADERS += src/modules/musig/keyagg_impl.h
